shared/common-adapters/markdown/shared.js

```diff
@@ -78,6 +80,21 @@ function createChannelRegex(meta: ?MarkdownMeta): ?RegExp {
   return new RegExp(`^#(${meta.mentionsChannelName.keySeq().join('|')})\\b`)
 }
 
+function createKbfsPathRegex(): ?RegExp {
+  const username = `([a-zA-Z0-9]+_?)+` // from go/kbun/username.go
+  const usernames = `${username}(,${username})*`
+  const teamName = `${username}(\\.${username})*`
+  const tlfType = `/(private|public|team)`
+  const tlf = `/((private|public)/${usernames}(#${usernames})?|team/${teamName})`
+  const inTlf = `/(\\S|\\\\\\\\|\\\\ )*[^/\\s]+` // don't include any trailing slash
+  return new RegExp(`^/keybase((${tlf}(${inTlf})?)|(${tlfType}))?`)
+}
+
+const kbfsPathRegex = createKbfsPathRegex()
+
+const getKbfsPathRegex = (meta: ?MarkdownMeta): ?RegExp =>
```

Comment:
Kill this, it shouldn't work this way. It should work like how channel works

Reply:
The channel thing calls `createChannelRegex` in `match`, which creates a new RegExp. Doesn't it result creating new objects all the time?

Reply:
Oh I see what you're saying now. It'd still be the same object.

Reply:
That's a good point, we could probably optimize it by doing the regex creation just ones inside the markdown meta (since the channel regex is dynamic)

Reply:
Sounds like we should do the same for other stuff (channel mentions, etc.) here too, So we don't create the objects all the time. Is this what you mean?